lightly/utils/scheduler.py

```diff
@@ -28,7 +36,7 @@ def cosine_schedule(
         raise ValueError("Current step number can't be negative")
     if max_steps < 1:
         raise ValueError("Total step number must be >= 1")
-    if step > max_steps:
+    if not period and step > max_steps:
```

Comment:
```suggestion
    if period is None and step > max_steps:
```

We usually check directly for `None` to avoid issues when the variable is `0` or another value that returns `False` when converted to boolean.